cmd/skopeo/completions.go

```diff
@@ -1,13 +1,57 @@
 package main
 
 import (
+	"github.com/containers/image/v5/directory"
+	"github.com/containers/image/v5/docker"
+	dockerArchive "github.com/containers/image/v5/docker/archive"
+	ociArchive "github.com/containers/image/v5/oci/archive"
+	oci "github.com/containers/image/v5/oci/layout"
+	"github.com/containers/image/v5/sif"
 	"github.com/containers/image/v5/tarball"
 	"github.com/containers/image/v5/transports"
 	"github.com/spf13/cobra"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
 )
 
-// autocompleteSupportedTransports list all supported transports with the colon suffix.
-func autocompleteSupportedTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+func autocompleteTransports(cmd *cobra.Command, args []string, toComplete string) ([]cobra.Completion, cobra.ShellCompDirective) {
+	directive := cobra.ShellCompDirectiveNoSpace
+	// We still don't have a transport
+	transport, details, haveTransport := strings.Cut(toComplete, ":")
+	if !haveTransport {
+		transports := supportedTransportSuggestions()
+		return transports, directive | cobra.ShellCompDirectiveNoFileComp
+	}
+	if transport == ociArchive.Transport.Name() || transport == dockerArchive.Transport.Name() || transport == sif.Transport.Name() || transport == oci.Transport.Name() {
+		return nil, directive
+	}
+	if transport == docker.Transport.Name() && details == "" {
+		return []cobra.Completion{docker.Transport.Name() + "://"}, directive | cobra.ShellCompDirectiveNoFileComp
```

Comment:
`transport+"://"` please: we’ve had a proposal to support other names for this transport, so let’s not hard-code more than necessary.